pkg/plugin/resources.go

```diff
@@ -467,6 +469,107 @@ func (a *App) handleLLMState(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+// provisionedPlugin is the response returned by a call to grafana.com's provisioned-plugin's endpoint
+// for an specific instance's plugin.
+type provisionedPlugin struct {
+	ID int `json:"id"`
+}
+
+// getPluginID gets the *ID* of the *provisioned plugin* from grafana.com.
+// Note that this differs to the plugin ID referred to by the `backend.CallResourceRequest`,
+// which is 'grafana-ml-app'
+func getPluginID(ctx context.Context, slug string, grafanaAppURL string, saToken string, gcomAPIKey string) (int, error) {
+	gcomPath := "/api/gnet/instances/" + slug + "/provisioned-plugins/grafana-llm-app"
+	req, err := http.NewRequestWithContext(ctx, "GET", grafanaAppURL+gcomPath, nil)
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", saToken))
+	req.Header.Set("X-Api-Key", gcomAPIKey)
+	if err != nil {
+		return 0, fmt.Errorf("create http request: %w", err)
+	}
+	respBody, err := doRequest(req)
+	if err != nil {
+		return 0, err
+	}
+	plugin := provisionedPlugin{}
+	err = json.Unmarshal(respBody, &plugin)
+	if err != nil {
+		return 0, fmt.Errorf("unmarshal json: %w", err)
+	}
+	return plugin.ID, nil
+}
+
+func (a *App) handleSavePluginSettings(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		handleError(w, fmt.Errorf("method not allowed: %s", req.Method), http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Read the request body
+	if req.Body == nil {
+		log.DefaultLogger.Warn("Request body is nil")
+		handleError(w, errors.New("request body required"), http.StatusBadRequest)
+		return
+	}
+
+	log.DefaultLogger.Debug("Getting provisioned plugin ID from grafana.com")
+
+	// The HG_INSTANCE_SLUG is only set for Grafana Cloud instances.
+	hgSlug := os.Getenv("HG_INSTANCE_SLUG")
+	var pluginID int
+	var err error
+	if hgSlug != "" && a.settings.EnableGrafanaManagedLLM {
+		pluginID, err = getPluginID(req.Context(), a.settings.Tenant, a.grafanaAppURL, a.saToken, a.settings.GrafanaComAPIKey)
+		if err != nil {
+			handleError(w, fmt.Errorf("get plugin ID: %w", err), http.StatusInternalServerError)
+			return
+		}
+	} else {
+		log.DefaultLogger.Info("Hosted Grafana Slug not found or plugin not provisioned; skipping getting provisioned pluginID")
+		pluginID = 0
```

Comment:
Could we write success and return here rather than requiring a second check of the setting and environment variable below?

If so typical go practice would be to invert the logic to do the if check and bail to avoid as much nesting of if statements.
```
if hgSlug == "" || !a.settings.EnableGrafanaManagedLLM {
    w.WriteHeader(http.StatusOK)
    _, _ = w.Write([]byte(`{"status": "success"}`))
    return
}
...